src/common/latest/sonarqube/Endpoint.ts

```diff
@@ -31,41 +30,38 @@ export default class Endpoint {
     this.data = data;
     // Remove trailing slash at the end of the base url, if any
     if (this.data) {
-      this.data.url = this.data?.url.replace(/\/$/, "");
+      this.data.url = this.data.url.replace(/\/$/, "");
     }
   }
 
   public get url() {
     return this.data.url;
   }
 
-  public get auth(): { username: string; password: string } {
+  public get auth(): { username: string; password?: string } {
     // If using user/password
     if (!this.data.token && this.data.password && this.data.password.length > 0) {
       return { username: this.data.username, password: this.data.password };
     }
     // Using token
-    return { username: this.data.token || this.data.username, password: "" };
+    return { username: this.data.token || this.data.username };
   }
 
-  toFetchOptions(endpointUrl: string, signal?: AbortSignal): Partial<RequestInit> {
-    const options: Partial<RequestInit> = {
-      method: "get",
-      signal,
-    };
-
-    // Add HTTP auth from this.auth
-    options.headers = {
-      Authorization:
-        "Basic " + Buffer.from(`${this.auth.username}:${this.auth.password}`).toString("base64"),
+  toAxiosOptions(): AxiosRequestConfig {
```

Comment:
Passing the endpoint URL doesn't really makes sense as we can directly access `this.url`

Reply:
the only difference is that we were passing "fullUrl" that included path + query, however, I don't think we need it